website_require_login/models/ir_http.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Advitus MB
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0).
+from pathlib import Path
+
+from odoo import SUPERUSER_ID, api, models
+from odoo.http import request
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = "ir.http"
+
+    @classmethod
+    def _dispatch(cls):
+        res = super(IrHttp, cls)._dispatch()
+
+        # if not website request - skip
+        env = api.Environment(request.cr, SUPERUSER_ID, request.context)
+        website = env["website"].get_current_website()
+        if not website:
+            return res
+
+        auth_paths = (
```

Comment:
it is better to search after checking `if request.uid == website.user_id.id`, so as not to search every time

Reply:
Fixed